sound/soc/sof/topology.c

```diff
@@ -1693,16 +1693,22 @@ int sof_load_pipeline_ipc(struct device *dev,
 			  struct sof_ipc_comp_reply *r)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
-	int ret = sof_core_enable(sdev, pipeline->core);
+	int ret;
+
+	mutex_lock(&sdev->cores_status_mutex);
```

Comment:
This should already be holding a DAPM lock since its part of the topology loading ?